src/validations/product.validation.ts

```diff
@@ -20,4 +20,58 @@ export const productSchema = {
     discount: Joi.number().min(0).max(100).allow(null),
     expiry_date: Joi.date().iso().allow(null),
   }),
+  update: Joi.object({
+    name: Joi.string().when('$name', {
+      is: Joi.exist(),
+      then: Joi.string().required(),
+    }),
+    description: Joi.string().when('$description', {
+      is: Joi.exist(),
+      then: Joi.string().required(),
+    }),
+    category_name: Joi.string().when('$category_name', {
+      is: Joi.exist(),
+      then: Joi.string().required(),
+    }),
+    price: Joi.number().when('$price', {
+      is: Joi.exist(),
+      then: Joi.number().required(),
+    }),
+    quantity: Joi.number().when('$quantity', {
+      is: Joi.exist(),
+      then: Joi.number().required(),
+    }),
+    image_url: Joi.array()
+      .items(Joi.string().uri())
+      .min(4)
+      .max(8)
+      .when('$image_url', {
+        is: Joi.exist(),
+        then: Joi.array().items(Joi.string().uri()).min(4).max(8).required(),
+      }),
+    discount: Joi.number()
+      .min(0)
+      .max(100)
+      .allow(null)
+      .when('$discount', {
+        is: Joi.exist(),
+        then: Joi.number().min(0).max(100).allow(null),
+      }),
+    expiry_date: Joi.date()
+      .iso()
+      .allow(null)
+      .when('$expiry_date', {
+        is: Joi.exist(),
+        then: Joi.date().iso().allow(null),
+      }),
+  }).or(
+    'name',
+    'description',
+    'category_name',
+    'price',
+    'quantity',
+    'image_url',
+    'discount',
+    'expiry_date',
+  ).error(new Error('At least one attribute required to update product'))
```

Comment:
Make research and look into the best between what you wrote and this. If yours is best then we can proceed with it. 
```suggestion
  }).min(1)
```

Reply:
ok let work on it

Reply:
yeah it works